src/util.h

```diff
@@ -494,14 +494,21 @@ class BufferValue : public MaybeStackBuffer<char> {
 // silence a compiler warning about that.
 template <typename T> inline void USE(T&&) {}
 
-// Run a function when exiting the current scope.
-struct OnScopeLeave {
-  std::function<void()> fn_;
-
-  explicit OnScopeLeave(std::function<void()> fn) : fn_(std::move(fn)) {}
-  ~OnScopeLeave() { fn_(); }
+template <typename Fn>
+struct OnScopeLeaveImpl {
+  Fn fn_;
+  ~OnScopeLeaveImpl() { fn_(); }
 };
 
+// Run a function when exiting the current scope. Used like this:
+// auto on_scope_leave = OnScopeLeave([&] {
+//   // ... run some code ...
+// });
+template <typename Fn>
+inline OnScopeLeaveImpl<Fn> OnScopeLeave(Fn&& fn) {
```

Comment:
is it worth adding an `__attribute__((unused))` here?

Reply:
@devsnek Yes – thank you for pointing that out!